integrations/coinbase/src/main/java/org/dash/wallet/integration/coinbase_integration/ui/CoinbaseServicesFragment.kt

```diff
@@ -75,40 +69,30 @@ class CoinbaseServicesFragment : Fragment(R.layout.fragment_coinbase_services) {
         }
 
         binding.buyDashBtn.setOnClickListener {
-           lifecycleScope.launch {
-                    sharedViewModel.paymentMethodsUiState.collect { uiState ->
-                        // New value received
-                        when (uiState) {
-                            is PaymentMethodsUiState.Success -> {
-                                uiState.paymentMethodsList.toTypedArray().let { paymentMethodsArray ->
-                                    CoinbaseServicesFragmentDirections.servicesToBuyDash(paymentMethodsArray)
-                                }.let { navDirection -> safeNavigate(navDirection) }
-                            }
-                            is PaymentMethodsUiState.Error -> {
-                                if (uiState.isError) {
-                                    AdaptiveDialog.create(
-                                        R.drawable.ic_info_red,
-                                        getString(R.string.coinbase_dash_wallet_no_payment_methods_error_title),
-                                        getString(R.string.coinbase_dash_wallet_no_payment_methods_error_message),
-                                        getString(R.string.close),
-                                        getString(R.string.add_payment_method),
-                                    ).show(requireActivity()) { addMethod ->
-                                        if (addMethod == true) {
-                                            viewModel.logEvent(AnalyticsConstants.Coinbase.BUY_ADD_PAYMENT_METHOD)
-                                            openCoinbaseWebsite()
-                                        }
-                                    }
-                                }
-                            }
-                            is PaymentMethodsUiState.LoadingState ->{
-                                if (uiState.isLoading) {
-                                    showProgress(R.string.loading)
-                                } else
-                                    dismissProgress()
-                            }
+            sharedViewModel.paymentMethodsUiState.observe(viewLifecycleOwner) { uiState ->
+                // New value received
+                when (uiState) {
+                    is PaymentMethodsUiState.Success -> {
+                        val paymentMethodsArray = uiState.paymentMethodsList.toTypedArray()
+
+                        if (paymentMethodsArray.isEmpty()) {
+                            showNoPaymentMethodsError()
```

Comment:
If `paymentMethodsList` is empty, this is functionally the same as an error case later.